textpattern/publish/taghandlers.php

```diff
@@ -3600,33 +3598,20 @@ function image_info($atts)
     $validItems = array('id', 'name', 'category', 'category_title', 'alt', 'caption', 'ext', 'author', 'w', 'h', 'thumb_w', 'thumb_h', 'date');
     $type = do_list($type);
 
-    $from_form = false;
-
-    if ($id) {
-        $thisimage = imageFetchInfo('id = '.intval($id));
-    } elseif ($name) {
-        $thisimage = imageFetchInfo("name = '".doSlash($name)."'");
-    } else {
-        assert_image();
-        $from_form = true;
-    }
-
     $out = array();
-    if ($thisimage) {
-        $thisimage['category_title'] = fetch_category_title($thisimage['category'], 'image');
+    if ($imageData = imageFetchInfo($id, $name)) {
+        $imageData['category_title'] = fetch_category_title($imageData['category'], 'image');
 
         foreach ($type as $item) {
             if (in_array($item, $validItems)) {
-                if (isset($thisimage[$item])) {
+                if (isset($imageData[$item])) {
                     $out[] = ($escape == 'html') ?
-                        txpspecialchars($thisimage[$item]) : $thisimage[$item];
+                        txpspecialchars($imageData[$item]) : $imageData[$item];
                 }
+            } else {
+                trigger_error(gTxt('invalid_attribute_value', array('{name}' => $item)), E_USER_NOTICE);
             }
         }
-
-        if (!$from_form) {
-            $thisimage = '';
```

Comment:
What it was, I reckon, was carelessness: mucking with a global instead of creating a local copy. Obviously `images()` needs to do this to pass data to the other image tags it contains. Equally obviously a tag such as `image_info()`, which cannot be used as a container, has no business changing the value of `$thisimage`.

And I'm really glad you brought this up, because `image_url()` can be used as a container. Maybe there's a valid use case for having it temporarily change `$thisimage`. For example, within an `images()` tag, `image_url()` looks for an image with name matching `$thisimage['author']`. And then the `image_url()` tag, also a container, wants the caption of that image of the author. A bit convoluted, perhaps, but not unreasonable. I'll add a commit to restore this behavior while we think it over.